### ISSUE TITLE
Python doc mentions `.append` for models and arrays but this raise `AttributeError: 'builtins.ReadOnlyRustModel' object has no attribute 'append'`

### ISSUE THREAD

**Issue by microhobby** [slint-ui/slint#8419]
### Bug Description

I have the following Slint code:

```js
export component AppWindow inherits Window {
    in-out property<[string]> SlintList : [];
}
```

And the following Python code:

```python
class App(app_components.AppWindow):

    def __init__(self):
        super().__init__()
        self.SlintList = slint.ListModel([])
```

But when I try to add a new element:

```
self.SlintList.append("test")
```

I get the error: `AttributeError: 'builtins.ReadOnlyRustModel' object has no attribute 'append'`
Looking at the code seems that the `ListModel` implements the `append` but still it does not work. I'm confused, let me know if I'm doing something wrong.

### Reproducible Code (if applicable)

```slint

```

### Environment Details

- Slint Version: 1.9.2
- Platform/OS: Debian/Linux
- Programming Language: Python
- Backend/Renderer: 

obs: Not using the latest Slint because on Debian stable there is no Python 3.12 yet, and Slint depends on it.

### Product Impact

There is a workaround, like use a fallback list and everytime you append to this fallback list you create a new `ListModel`.

**Comment by tronical** [comment]
I can't reproduce this :-(. Our own printer demo uses this at it works (well, it requires this patch to truly trigger the same)

```patch
diff --git a/demos/printerdemo/python/main.py b/demos/printerdemo/python/main.py
index c7aeba4c2..0e211bb62 100644
--- a/demos/printerdemo/python/main.py
+++ b/demos/printerdemo/python/main.py
@@ -38,7 +38,7 @@ class MainWindow(slint.loader.ui.printerdemo.MainWindow):

     @slint.callback(global_name="PrinterQueue", name="start_job")
     def push_job(self, title):
-        self.printer_queue.append(
+        self.PrinterQueue.printer_queue.append(
             PrinterQueueItem(
                 status="waiting",
                 progress=0,
```

Could you try to modify one of our examples to reproduce or point me to a repo that reproduce it?

I also tried this unit test, that passes for me:

```patch
diff --git a/api/python/tests/test_models.py b/api/python/tests/test_models.py
index 7b3c2e7a6..973d44a86 100644
--- a/api/python/tests/test_models.py
+++ b/api/python/tests/test_models.py
@@ -157,3 +157,33 @@ def test_model_writeback() -> None:
     assert list(instance.get_property("model")) == [100, 42]
     instance.invoke("write-to-model", 0, 25)
     assert list(instance.get_property("model")) == [25, 42]
+
+
+def test_model_cast() -> None:
+    compiler = native.Compiler()
+
+    compdef = compiler.build_from_source(
+        """
+  export component App {
+    width: 300px;
+    height: 300px;
+
+    in-out property<[int]> model;
+
+  }
+    """,
+        Path(""),
+    ).component("App")
+    assert compdef is not None
+
+    instance = compdef.create()
+    assert instance is not None
+
+    model = models.ListModel([100, 0])
+
+    instance.set_property("model", model)
+
+    model_clone = instance.get_property("model")
+    model_clone.append(42)
+
+    assert list(model) == [100, 0, 42]
```

**Comment by microhobby** [comment]
@tronical thanks to present `uv` it's pretty cool! And now using a newer version of Python and latest slint package still getting the same behavior. The repo is this one: https://github.com/gaiaBuildSystem/phobosEmulatorManager

**Comment by microhobby** [comment]
Ok, so the behaviors from here and https://github.com/slint-ui/slint/issues/8418 are fixed by setting the model out of the App constructor: https://github.com/gaiaBuildSystem/phobosEmulatorManager/commit/fa7df8843c042ae62ea774999d255223be9a04d6

Seems that the model was not fully initialized yet 🤔
For me is ok to set it after, so closing this, thanks a lot for the insights and tips @tronical
